src/components/Pagination.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+const Pagination = ({ article, pageNumber }) => {
+  const activePage = number => (number === Number(pageNumber) ? 'active' : null);
+  const disablePrevious = (num1, num2) => (num1 === num2 ? 'disabled' : null);
+
+  const numberOfpages = Math.ceil(article.count / 5);
+  const ArraynumberOfpages = _.range(1, numberOfpages + 1);
+  const pages = ArraynumberOfpages.map(number => (
+    <li key={number} className={`page-item ${activePage(number)}`}>
+      {' '}
+      <a className="page-link" href={`articles/page/${number}`}>
```

Comment:
Since we are building an SPA, you might have to use the `<NavLink>` from `react-router-dom` in order to retain application state from previous requests

Reply:
i didn't get you here, can you please elaborate more on this. Thank You

Reply:
Let me try to rephrase, since we are building an SPA (Single Page Application), an `anchor-tag` would do a full page reload, which means  booting the entire  application, refetching all the  `assets` (`css`, `js`)  and rendering the UI that the user  has already done. 

What we want instead is to fetch the missing  parts from the server without re-rendering the whole application again. 
```jsx
<NavLink className="page-link" to={`articles/page/${number}`}/>
```

Reply:
I believe that a descriptive naming would do in this case, instead of using `num1, num2`...